Gemfile

```diff
@@ -3,6 +3,7 @@ git_source(:github){|repo| "https://github.com/#{repo}.git"}
 
 ruby "2.7.0"
 
+gem "bcrypt", "3.1.13"
```

Comment:
gem này để làm gì bạn

Reply:
Gem bcrypt là gem sử dụng thuật toán mã hóa Blowfish để mã hóa mật khẩu thành chuỗi password_digest, đây là mã hóa một chiều, nó tự động tạo ra các chuỗi mã hóa ngẫu nhiên sau mỗi lần tạo ra.Điều này làm cho nó khó khăn hơn cho việc mã hóa và tìm ra mật khẩu cho những cuộc tấn công. Nó cũng có thể được thực hiện để đi chậm hơn, giúp chống lại các cuộc tấn công. Bcrypt hỗ trợ đầu vào của mỗi ký tự ở định dạng UTF-8, vì vậy nó có thể băm mật khẩu mà không giới hạn ký tự cho các ký tự cả chữ và số